### ISSUE TITLE
Remove blank line under a 'none' content block will remove 'none' block together

### ISSUE THREAD

**Issue by tcper** [TypeCellOS/BlockNote#605]
**Describe the bug**
When I need to remove a blank line under a 'none' type content, it never works, remove the 'none' content together.
I think remove a blank line is reasonable action.
![20240305-161541](https://github.com/TypeCellOS/BlockNote/assets/363082/97ac9df9-264b-468e-9361-e3cb94850e27)
```
const blockConfig = {
  type: "image" as const,
  propSchema: imagePropSchema,
  content: "none",
} satisfies CustomBlockConfig;
```
I tried, all `content: "none",` will cause this bug.

**To Reproduce**
1. Insert a 'none' content block, like 'image'
2. Insert a blank line under the 'none' content block
3. Press Backspace in the blank line

**Misc**
- Node version:
- Package manager:
- Browser: 
- [ ] I'm a [sponsor](https://github.com/sponsors/YousefED) and would appreciate if you could look into this sooner than later 💖

**Comment by sb8244** [comment]
Is the solution here to check if the block is paragraph, has no content, and focus is start of block (should be impossible to meet these criteria and not meet this one). If that matches, then delete the block rather than merging it?

**Comment by matthewlipski** [comment]
Yeah that sounds right - we should already have logic [here](https://github.com/TypeCellOS/BlockNote/blob/280c280600823a6b432bbaca6e4290f8c03d8b91/packages/core/src/pm-nodes/BlockContainer.ts#L500) for that, but seems like there's a bug in there somewhere

**Comment by matthewlipski** [comment]
Hmm have you tried modifying the code from `BNMergeBlocks`? I think if you just remove the `.replace` [here](https://github.com/TypeCellOS/BlockNote/blob/908cf28e13e7a601b2713518056345cccdd1a6c2/packages/core/src/pm-nodes/BlockContainer.ts#L396), you should get pretty close to the correct behaviour. The difference is that `BNMergeBlocks` will just un-nest the children of the later block, rather than attaching them to the previous one which might still be a dealbreaker. Either way I think it's smth you might want to look into

**Comment by lamtuanamc** [comment]
> @matthewlipski I'm not sure any of the code blocks there handle that cases this. At least when I looked into it, I came to that conclusion.
> 
> I wrote some custom keydown handlers (on `BlockNoteView`) that handle this case and another. One big thing here is pulling the children from the block to the prevBlock:
> 
> ```ts
> if (e.key === "Backspace") {
>           const selection = blockNoteEditor.getSelection()
>           const position = blockNoteEditor.getTextCursorPosition()
> 
>           if (selection === undefined && position && position.block.type === "paragraph") {
>             const block = position.block
>             const blockPos = getBlockInfoFromPos(
>               blockNoteEditor._tiptapEditor.state.doc,
>               blockNoteEditor._tiptapEditor.state.selection.anchor
>             )
> 
>             const isTopBlock = blockPos.depth === 2
>             const isEmptyBlock = block.content.length === 0 && block.children.length === 0
>             const isEmptyContentWithChildren = block.content.length === 0 && block.children.length > 0
> 
>             if (isTopBlock && isEmptyBlock) {
>               e.stopPropagation()
>               e.preventDefault()
>               blockNoteEditor.removeBlocks([block])
> 
>               if (position.prevBlock) {
>                 blockNoteEditor.setTextCursorPosition(position.prevBlock, "end")
>               }
>             } else if (isTopBlock && position.prevBlock && isEmptyContentWithChildren && Array.isArray(position.prevBlock.children)) {
>               e.stopPropagation()
>               e.preventDefault()
>               blockNoteEditor.updateBlock(position.prevBlock, { children: position.prevBlock.children.concat(block.children) })
>               blockNoteEditor.removeBlocks([block])
> 
>               if (position.prevBlock) {
>                 blockNoteEditor.setTextCursorPosition(position.prevBlock, "end")
>               }
>             }
>           }
>         }
> ```

@sb8244  Can you give me the definition of the function getBlockInfoFromPos?

**Comment by lamtuanamc** [comment]
Hi @matthewlipski, I've created a pull request to resolve this issue. Could you please review it at your convenience? Thank you!"
